src/Reflection/ReflectionClass.php

```diff
@@ -370,22 +380,17 @@ public function getEndLine()
      * You may optionally specify a source locator that will be used to locate
      * the parent class. If no source locator is given, a default will be used.
      *
-     * @param SourceLocator|null $sourceLocator
      * @return ReflectionClass
      */
-    public function getParentClass(SourceLocator $sourceLocator = null)
+    public function getParentClass()
     {
         if (null === $this->extendsClassType) {
             return null;
         }
 
-        $fqsen = $this->extendsClassType->__toString();
-
-        if (null !== $sourceLocator) {
-            $parent = (new ClassReflector($sourceLocator))->reflect($fqsen);
-        } else {
-            $parent = self::createFromName($fqsen);
-        }
+        // @TODO use actual `ClassReflector` or `FunctionReflector`?
+        /* @var $parent self */
+        $parent = $this->reflector->reflect((string) $this->extendsClassType);
```

Comment:
Can you not just call `$this->reflectClassForNamedNode((string)$this->extendsClassType);`?